NetworkLayer/Model/CoreAdvertsService.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  CoreAdvertsService.swift
+//  NetworkLayer
+//
+//  Created by Clément Garcia on 17/10/2022.
+//
+
+import Foundation
+
+/// Handle the services related to adverts
+final class CoreAdvertsService {
+    
+    // MARK: - Vars
+    /// URLSession handling the call to the REST apis
+    private let session: URLSession
+    /// Service to handle the categories used within adverts
+    private let categoryService = CoreCategoryService()
+    
+    // MARK: - initializer
+    init(session: URLSession = .shared) {
+        self.session = session
+    }
+    
+    // MARK: - Function
+    /// Retreive the list of adverts
+    /// - Returns: List of adverts
+    func getAdvertsList() async throws -> AdvertsList {
+        let (data, _) = try await session.data(from: ApiEndpoint.getEndpointAdvert())
```

Comment:
```suggestion
        let (data, response) = try await session.data(from: ApiEndpoint.getEndpointAdvert())
```

Si tu veux être complet, tu dois vérifier la réponse ici.